api/converter_utils.go

```diff
@@ -407,6 +408,99 @@ func signedTxnWithAdToTransaction(stxn *transactions.SignedTxnWithAD, extra rowD
 		}
 
 		application = &a
+	case protocol.StateProofTx:
+		partPath := make([]generated.GenericDigest, len(stxn.Txn.StateProof.PartProofs.Path))
+		for idx, part := range stxn.Txn.StateProof.PartProofs.Path {
+			partPath[idx] = generated.GenericDigest(part)
```

Comment:
I believe you need to make a copy of `part` because the pointer is reused by the for loop.

Reply:
Here's an example where we do something similar: https://github.com/algorand/indexer/pull/575/files